java/core/src/main/java/sleeper/core/util/IteratorFactory.java

```diff
@@ -50,23 +58,51 @@ public IteratorFactory(ObjectFactory inner) {
+            if (iteratorConfig.getFilters() == null) {
+                ConfigStringIterator iterator;
+                String className = iteratorConfig.getIteratorClassName();
 
-            // If aggregation keyword is used, create specific iterator
-            if (className.equalsIgnoreCase(DataEngine.AGGREGATION_ITERATOR_NAME)) {
-                iterator = new ConfigStringAggregationFilteringIterator();
+                // If aggregation keyword is used, create specific iterator
+                if (className.equalsIgnoreCase(DataEngine.AGGREGATION_ITERATOR_NAME)) {
+                    iterator = new ConfigStringAggregationFilteringIterator();
+                } else {
+                    iterator = inner.getObject(className, ConfigStringIterator.class);
+                }
+                LOGGER.debug("Created iterator of class {}", className);
+                iterator.init(iteratorConfig.getIteratorConfigString(), iteratorConfig.getSchema());
+                LOGGER.debug("Initialised iterator with config {}", iteratorConfig.getIteratorConfigString());
+                return iterator;
             } else {
-                iterator = inner.getObject(className, ConfigStringIterator.class);
+                AggregationFilteringIterator iterator = new AggregationFilteringIterator();
+                iterator.setFilterAggregationConfig(getConfigFromProperties(iteratorConfig));
+                iterator.setSchema(iteratorConfig.getSchema());
+                return iterator;
             }
-            LOGGER.debug("Created iterator of class {}", className);
-            iterator.init(iteratorConfig.getIteratorConfigString(), iteratorConfig.getSchema());
-            LOGGER.debug("Initialised iterator with config {}", iteratorConfig.getIteratorConfigString());
-            return iterator;
         } catch (ObjectFactoryException exc) {
             throw new IteratorCreationException(exc);
         }
     }
+
+    /**
+     * Gets the filter aggregation config when set by table properties.
+     * Currently just gets the filter ageoff(column,age).
+     * Currently only uses filters from input but eventually will build aggregations from it too.
+     *
+     * @param  iteratorConfig config for an iterator that should have filters set in it
+     * @return                filter aggregation config to be used in an iterator
+     */
+    private FilterAggregationConfig getConfigFromProperties(IteratorConfig iteratorConfig) {
+        String[] filterParts = iteratorConfig.getFilters().split("\\(");
+        if ("ageoff".equals(filterParts[0].toLowerCase(Locale.ENGLISH))) {
+            String[] filterInput = StringUtils.chop(filterParts[1]).split(","); //Chop to remove the trailing ')'
+            Optional<String> filterColumn = Optional.of(filterInput[0]);
+            long maxAge = Long.parseLong(filterInput[1]);
+            return new FilterAggregationConfig(List.of("TODO in 5477"), filterColumn, maxAge, List.of());
```

Comment:
Because the required value fields include values that are not real field names, I think code that uses this will break. It would be good if we can test it all the way through. There are some tests that do this already, and we could update those to test against the new configuration method:

IngestRowsIT (currently there's shouldApplyIterator, we could add another one)
JavaCompactionRunnerIteratorIT
QueryExecutorTest.ApplyIterators